### PR TITLE
2.x: Reuse SingleDoOnEvent

### PR DESCRIPTION
**Pull request by vanniktech**

### CODE REVIEW

File: src/main/java/io/reactivex/Single.java [hunk 1]
```diff
@@ -1588,7 +1588,7 @@
      */
     public final Single<T> doOnSuccess(final Consumer<? super T> onSuccess) {
         ObjectHelper.requireNonNull(onSuccess, "onSuccess is null");
-        return RxJavaPlugins.onAssembly(new SingleDoOnSuccess<T>(this, onSuccess));
+        return RxJavaPlugins.onAssembly(new SingleDoOnEvent<T>(this, onSuccess, Functions.emptyConsumer()));
```
Comment (akarnokd) [inline src/main/java/io/reactivex/Single.java:1591]: It was inlined nicely, this now allocates 2 consumers.

Comment (vanniktech) [reply src/main/java/io/reactivex/Single.java:1591]: True I just thought of reusing it since that's also the way it is done in Completable

Comment (akarnokd) [reply src/main/java/io/reactivex/Single.java:1591]: Single and Completable are practically a year old and although re-architected, they still run with extra allocations. If I can get there eventually, I'll tidy them up.

Comment (vanniktech) [reply src/main/java/io/reactivex/Single.java:1591]: Alright cool then I'll close this and help you out with avoiding the extra costs of allocation

**Pull request closed without merging**